realsense_ros_camera/include/base_realsense_node.h

```diff
@@ -109,6 +113,8 @@ namespace realsense_ros_camera
         std::map<stream_index_pair, bool> _enable;
         std::map<stream_index_pair, std::string> _stream_name;
         tf2_ros::StaticTransformBroadcaster _static_tf_broadcaster;
+        tf::TransformBroadcaster _dynamic_tf_broadcaster;
+        boost::shared_ptr<boost::thread> _transform_thread;
```

Comment:
This pointer doesn't appear to be shared anywhere. A std::unique_ptr might be better. (This object doesn't seem to be referenced anywhere, so the difference in access time and overhead is negligible)